### PR TITLE
Show event time on events page

### LINKED ISSUE #2676: same events need to show time
### Feature description

There is not way to attach a time to events in https://qiskit.org/events/ 

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by y4izus**
## Changes

This PR makes the necessary changes to the codebase to be able to display an event's start time if defined.

Closes #2676 

Co-authored-by: @eddybrando 

## Implementation details

The structure of the Airtable events database has changed. A field for the "start date and time" of an event record is now available.

This introduces a new level of complexity while sorting the records. Before, we could use the sorting parameter from the Airtable `select` method to get the events sorted. Now, since we have 2 fields that are relevant for the sorting of the events, we need a custom implementation.

For this, a method called `sortEventsByStartDate` has been created in **hooks/event-conversion-utils.ts** and is responsible for the sorting of the events.

In addition to this, other methods have been added to support the formatting of the date and time data.

Also, some unnecessary methods have been removed and the tests updated.

## How to read this PR

- Start with the simple changes to **pages/events/index.vue** and **store/events.ts**
- Check the new `sortEventsByStartDate` method in **hooks/event-conversion-utils.ts**
- Review the simplified `getEventsQuery` method in **hooks/event-conversion-utils.ts** and the related changes in `fetchCommunityEvents` and `fetchSeminarSeriesEvents`
- See the changes to the `convertToCommunityEvent` and `convertToSeminarSeriesEvent` methods that result in the removal of some unnecessary methods
- See the other changes to the data and time functions in **hooks/event-conversion-utils.ts**
- Check the updates to the tests in **tests/hooks/event-conversion-utils.spec.ts**

## Screenshots

Screenshot 2023-02-06 at 16 52 39

Screenshot 2023-02-06 at 16 52 46

### CODE REVIEW

File: hooks/event-conversion-utils.ts [hunk 1]
```diff
@@ -164,35 +195,49 @@ class EventsAirtableRecords extends AirtableRecords {
...
-      types: this.getTypes(record),
-      image: await this.getImage(record),
-      location: this.getLocation(record),
-      regions: this.getRegions(record),
+      endDate: record.get(this.recordFields!.endDate) || '',
+      location: record.get(this.recordFields!.location) || WORLD_REGIONS.tbd,
+      regions: record.get(this.recordFields!.regions) || [WORLD_REGIONS.tbd],
+      title: record.get(this.recordFields!.name) || '',
+      to: record.get(this.recordFields!.website) || '',
+
```
Comment (techtolentino) [inline hooks/event-conversion-utils.ts:214]: non-blocking question: 
is this empty line here for a reason?

File: hooks/event-conversion-utils.ts [hunk 2]
```diff
@@ -164,35 +195,49 @@ class EventsAirtableRecords extends AirtableRecords {
...
+   */
+  async convertToSeminarSeriesEvent (record: Record<string, any>): Promise<SeminarSeriesEvent> {
     const event = {
+      endDate: record.get(this.recordFields!.endDate) || '',
+      institution: record.get(this.recordFields!.institution) || '',
+      location: record.get(this.recordFields!.location) || WORLD_REGIONS.tbd,
+      speaker: record.get(this.recordFields!.speaker) || '',
+      title: record.get(this.recordFields!.name) || '',
+      to: record.get(this.recordFields!.website) || '',
+
```
Comment (techtolentino) [inline hooks/event-conversion-utils.ts:237]: same question as above... is there a reason for the empty line?

**Review by techtolentino — APPROVED** [review]
Really nice job here! The PR description + JSDocs helped a ton...

Just a non-blocking question... other than that, 🚀

File: hooks/event-conversion-utils.ts (hunk 1 above)
Comment (eddybrando) [reply hooks/event-conversion-utils.ts:214]: I was just trying to visually separate the properties that get their data using `record.get()` from the one using other more complex functions. Not really necessary, just thought that it helps when reading.

File: hooks/event-conversion-utils.ts (hunk 2 above)
Comment (eddybrando) [reply hooks/event-conversion-utils.ts:237]: Same as [here](https://github.com/Qiskit/qiskit.org/pull/2941/files/33ff37851d4ae9fb4ab0055e892b140ef654979f#r1098393713):

> I was just trying to visually separate the properties that get their data using `record.get()` from the one using other more complex functions. Not really necessary, just thought that it helps when reading.

Comment (techtolentino) [reply hooks/event-conversion-utils.ts:237]: sounds good @eddybrando 👍🏼

**Pull request merged**